src/Cli/Microsoft.DotNet.Cli.Utils/TelemetryEventEntry.cs

```diff
@@ -47,6 +48,28 @@ void Handler(object sender, InstrumentationEventArgs eventArgs)
         }
     }
 
+    public class PerformanceMeasurement : IDisposable
+    {
+        private readonly Stopwatch _timer;
+        private readonly Dictionary<string, double> _data;
+        private readonly string _name;
+
+        public PerformanceMeasurement(Dictionary<string, double> data, string name)
+        {
+            // Measurement is a no-op if we don't have a dictionary to store the entry.
+            if (data == null)
+            {
+                return;
+            }
+
+            _data = data;
+            _name = name;
+            _timer = Stopwatch.StartNew();
+        }
+
+        public void Dispose() => _data?.Add(_name, _timer.Elapsed.TotalMilliseconds);
```

Comment:
Should this call GC.SuppressFinalize() and Dispose(disposing)?

Reply:
I'm pretty sure that only applies when you have a finalizer, right? Like, if I had a `~PerformanceMeasurement()` method. I'll double check though to make sure.

Reply:
Discussed offline and sealed the class to avoid any potential issues.